src/rgw/rgw_rest_swift.cc

```diff
@@ -1943,9 +2003,9 @@ bool RGWFormPost::is_next_file_to_upload()
 
     const auto field_iter = part.fields.find("Content-Disposition");
     if (std::end(part.fields) != field_iter) {
-      const auto& params = field_iter->second.params;
+      auto& params = field_iter->second.params;
 
-      if (std::end(params) != params.find("filename")) {
+      if (std::end(params) != params.find("filename") && ! params["filename"].empty()) {
```

Comment:
The `operator[]` can unnecessarily modify the `params`. Maybe replace with something like that:
```cpp
      const auto& params = field_iter->second.params;
      const auto& filename_iter = params.find("filename");
      if (std::end(params) != filename_iter && ! filename_iter->empty()) {
```